### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

**Comment by fridex** [comment]
Discarding WIP flag, ready for review. There are still missing tests for some corner cases, I marked them with TODO. I would like to get this merged in as tests/changes I plan to do will change the behaviour of the existing codebase (hence better have it atomic).

File: tests/conftest.py [hunk 1]
```diff
@@ -17,9 +17,21 @@
...
 import pytest
+import flexmock
+
+from thoth.adviser.context import Context
+from thoth.adviser.beam import Beam
+from thoth.adviser.enums import RecommendationType
+from thoth.adviser.pipeline_config import PipelineConfig
+from thoth.adviser.predictor import Predictor
+from thoth.adviser.resolver import Resolver
+from thoth.adviser.predictor import Predictor
```
Comment (pacospace) [inline tests/conftest.py]: `from thoth.adviser.predictor import Predictor` twice

Comment (fridex) [reply tests/conftest.py]: Thanks! Addressed in 61c83eb
